### ISSUE TITLE
StandardScaler in DataFrameMapper throws DeprecationWarning (scikit-learn >= 0.17.0)

### ISSUE THREAD

**Issue by joshmorel** [paulgb/sklearn-pandas#67]
When using StandardScaler() in DataFrameMapper I get a DeprecationWarning when passing the column name as a string . I am warned of a ValueError in scikit-learn 0.19 although, I'm not yet getting this. Passing the name as a list avoids the issue, but I think this need should be hidden from the user. 

I can fix this myself in a fork but am kind of new to making pull requests so I want to post this as an issue first and confirm the desired workflow for doing this. 

Here is an example with output:

``` python

import pandas as pd
from sklearn.preprocessing import StandardScaler
from sklearn_pandas import DataFrameMapper

df = pd.DataFrame({'iq': [100,90,20,200], 'height': [140.1,130.7,170.1,100.34]})
mapper = DataFrameMapper([
        ('height', StandardScaler())
])
X = mapper.fit_transform(df)
```

``` python
/usr/local/lib/python3.5/dist-packages/sklearn/preprocessing/data.py:586: DeprecationWarning: Passing 1d arrays as data is deprecated in 0.17 and will raise ValueError in 0.19. Reshape your data either using X.reshape(-1, 1) if your data has a single feature or X.reshape(1, -1) if it contains a single sample.
  warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
/usr/local/lib/python3.5/dist-packages/sklearn/preprocessing/data.py:649: DeprecationWarning: Passing 1d arrays as data is deprecated in 0.17 and will raise ValueError in 0.19. Reshape your data either using X.reshape(-1, 1) if your data has a single feature or X.reshape(1, -1) if it contains a single sample.
  warnings.warn(DEPRECATION_MSG_1D, DeprecationWarning)
```

**Comment by dukebody** [comment]
Thanks taking the time to report this!

This issue is related to scikit-learn reworking the transformers interface. All arrays passed to "data" transformers (as opposed to "label" transformers like `LabelEncoder`) must be 2-d: Rows are considered examples/instances and columns are features. Previously, if a 1-d array was passed, it was not clear if it was one example with several features or one feature for several examples. This was causing issues with transformers like `StandardScaler` or `Imputer`.

We can modify sklearn-pandas to make
```
mapper = DataFrameMapper([
        ('height', StandardScaler())
])
```
and
```
mapper = DataFrameMapper([
        (['height'], StandardScaler())
])
```
equivalent. This is, string selectors are automatically transformed to list selectors under the hoods, so transformers _always_ receive 2d-arrays. However that will generate warnings for `LabelEncoder` and `LabelBinarizer`. The main reason I (and I believe other people) use `LabelEncoder` and `LabelBinarizer` in the pipeline is that `OneHotEncoder` doesn't allow string values, which are super common in real-world pandas datasets. See https://github.com/scikit-learn/scikit-learn/issues/4920.

For now I'd prefer to keep the possibility in sklearn-pandas to pass 1d-arrays to transformers. According to https://github.com/scikit-learn/scikit-learn/pull/7327 we should expect sklearn 0.19 to include a `OneHotEncoder` that can deal with string values, and therefore we can think of start deprecating string selectors here.

What do you think? Does this sound acceptable to you?

**Comment by joshmorel** [comment]
That's good. You can close this. Looking forward to the OneHotEncoder upgrade in scikit-learn!
